src/transxchange/naptan.rs

```diff
@@ -150,10 +150,26 @@ where
         .delimiter(b',')
         .trim(csv::Trim::All)
         .from_reader(reader);
+    let mut stop_points = CollectionWithId::default();
     for record in reader.deserialize() {
-        let _stop: Stop = record.with_context(|_| "Error parsing the CSV record into a Stop")?;
+        let stop: Stop = record.with_context(|_| "Error parsing the CSV record into a Stop")?;
+        let stop_area_id = stops_in_area.get(&stop.atco_code).cloned().ok_or_else(|| {
+            format_err!(
+                "Failed to find the corresponding StopArea for the StopPoint '{}'",
```

Comment:
you also can check that `stop_area_id` is in collections.stop_areas
but this could be done in `read_stops_in_area` instead of `validate_stops`

Reply:
I prepared the validation in a following PR (see the diff here https://github.com/woshilapin/transit_model/compare/naptan-stops...woshilapin:naptan-validate?expand=1)